apps/src/gamelab/GameLab.js

```diff
@@ -709,11 +709,7 @@ GameLab.prototype.rerunSetupCode = function() {
   ) {
     return;
   }
-  this.gameLabP5.p5.allSprites.removeSprites();
-  this.JSInterpreter.deinitialize();
-  this.initInterpreter(false /* attachDebugger */);
-  this.onP5Setup();
-  this.gameLabP5.p5.redraw();
+  this.execute(true /* forPreview */);
```

Comment:
It seems like there was an initial choice to only run a subset of the `execute` function during the implementation of preview: https://github.com/code-dot-org/code-dot-org/pull/21670/commits/07e1a9f1111445e7435675f9c5dfa749f907124b

Are there any side effects of changing this to the full `execute` function?

Reply:
Changed back. I'm not sure exactly why it was originally changed to only run a subset, but leaving it that way seems reasonable.